### DISCUSSION TITLE
Airflow crashes when we execute numerous AWS RdsCreateDbSnapshotOperator tasks within the same DAG.

### DISCUSSION THREAD

**Discussion by hzitoun-thefork** [apache/airflow#38124] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### If "Other Airflow 2 version" selected, which one?

v2.6.3

### What happened?

We are running Airflow on Kubernetes on AWS. When we have `19` operators of type `RdsCreateDbSnapshotOperator` airflow loses control, tasks retry, hang and fail, and we start to have zombi pods. With `18` operators everything works well.
I precise that we use a pool of `20 Slots`. 

Here is our args:

```python
create_pool_if_not_exist(
    pool_name="rds_snapshots_tasks", slots=20, description="pool for rds snapshot tasks"
)

operator = CustomRdsCreateDbSnapshotOperator(
      pool="rds_snapshots_tasks",
      task_id=f"collection_rds_{rds_name}_create_snapshot",
      db_type="instance",
      wait_for_completion=True,
      # Mandatory as true, NEVER set to false, it must remain in the pool while doing the snapshot
      dag=dag,
      aws_conn_id=f"aws_rds{specific_conn_identifier.lower()}",
      hook_params={"region_name": config_service.get_config_value("AWS_RDS_REGION")},
      db_identifier=db_identifier,
      db_snapshot_identifier=rds_export_task_identifier,  # max 100 characters on AWS
      priority_weight=rds_conf.get("priority_weight", 1),
      check_interval=60,
      max_attempts=2 * 60,
  )
```

### What you think should happen instead?

It should run without any issue, because there are a lot of free RAM and CPU in the namespace of airflow.

### How to reproduce

Create a dag with 19 operators of type RdsCreateDbSnapshotOperator inside. Run all the tasks at the same time. 

### Operating System

kubernetes

### Deployment

Official Apache Airflow Helm Chart

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Taragolis** [comment]
Seems like more troubleshooting rather than issue. Without additional information it is hard to say what is it actually. It would be nice to have some logs from the scheduler / worker, as well as versions of Amazon Provider and.. maybe other dependencies.

**Comment by hzitoun-thefork** [comment]
@Taragolis so the versions I use are:

```yaml
apache-airflow-providers-amazon = "==8.4.0"
apache-airflow-providers-cncf-kubernetes = "==7.4.0"
airflow-exporter = "==1.5.3"
apache-airflow = "==2.6.3"
```

we don't have logs actually. the task that fails doesn't start. All what we get the from task is this:

```python
XXXXXXXX-job-o
*** No logs found on s3 for ti=<TaskInstance:XXXXXXXX__2024-03-13T10:44:01.946670+00:00 [failed]>
*** Could not read served logs: [Errno -2] Name or service not known
```
Also the first task to fail and hang is very simple task `S3CreateObjectOperator`

And i want to mention that the tasks of `RdsCreateDbSnapshotOperator` don't even start and that makes airflow crashes and lose control. At certain point the entire DAG gets removed from the DAGBAG. 

We have other tasks/pods in the DAG (other than `RdsCreateDbSnapshotOperator`).
